### PR TITLE
Avoid multiple calls to getUserMedia for getLocalDevices

### PR DESCRIPTION
**Pull request by lukasIO**
fixes iOS safari video preview disappearing. 
Alternative fix on the SDK level to https://github.com/livekit/livekit-react/pull/60.
With this fix, we don't have to change anything in the react packages.

### CODE REVIEW

File: src/room/track/create.ts [hunk 1]
```diff
@@ -30,7 +31,15 @@ export async function createLocalTracks(
 
   const opts = mergeDefaultOptions(options, audioDefaults, videoDefaults);
   const constraints = constraintsForOptions(opts);
-  const stream = await navigator.mediaDevices.getUserMedia(constraints);
```
Comment (davidzhao) [inline src/room/track/create.ts:33]: this seems like it'd be problematic when we require different constraints? It seems that each attempt could be different.

**Review by davidzhao — COMMENTED** [review]
I wonder if there's an alternative way to trigger the permission dialog without impacting normal devices.

Is it worth trying this approach:
* in getDevices, calling navigator.mediaDevices.getUserMedia() but with constraints that wouldn't match any device.. for example, deviceId: "dummy".

Thinking out loud, the problem only exists if we acquire the same device twice, so if we could make it so that the getUserMedia attempt in discovery grants permissions, but fails to acquire an actual device.

**Comment by lukasIO** [comment]
nice idea! will test that approach with a dummy deviceId.

File: src/room/track/create.ts (hunk 1 above)
Comment (davidzhao) [reply src/room/track/create.ts:33]: ok, I totally misread the code.. it wasn't re-using it, but just caching the previous one.. clever!

this should work as well.

Comment (davidzhao) [reply src/room/track/create.ts:33]: though I think there are still some edge cases as it relates to audio/video types.

**Comment by lukasIO** [comment]
@davidzhao your suggestion works like a charm also on iOS. It's a great workaround!

**Review by davidzhao — APPROVED** [review]
nice!!

**Comment by lukasIO** [comment]
@davidzhao the map approach seems to work across browsers as long as the safari case is handled separately. tested across firefox/chrome/safari and iOS safari

**Review by davidzhao — APPROVED** [review]
awesome work!

**Pull request merged**